src/midje/emission/colorize.clj

```diff
@@ -2,40 +2,27 @@
             Midje output be ergonomically colorful."}
   midje.emission.colorize
   (:require [colorize.core :as color]
-            [clojure.string :as str]
-            [midje.config :as config]
-            [midje.util.ecosystem :as ecosystem]))
+            [midje.config :as config]))
 
 ;; This indirection is required so that the tests of this
 ;; file can fake the prerequisite
 (def config-choice config/choice)
 
-(defn- config-choice-as-string []
-  (let [choice (config-choice :colorize)]
-    (if (keyword? choice)
-      (name choice)
-      (str choice))))
+(defmacro ^:private def-colorize
```

Comment:
I personally don't like using macros when functions also suffice.
In this case it could be something like
```clojure
(defn- build-colorizer [normal-color reverse-color]
  (fn [a-str]
    (let [colorize-fn (case (config-choice :colorize)
                        :true normal-color
                        :reverse reverse-color
                        str)]
      (colorize-fn a-str))))

(def fail (build-colorizer color/red color/red-bg))
(def pass (build-colorizer color/green color/green-bg))
(def note (build-colorizer color/cyan color/cyan-bg))
```

Reply:
Good point, I'll refactor this.